### PR TITLE
Privileges bats first pass. Refactor `server_query` bats helper.

### PR DESCRIPTION
**Pull request by timsehn**
In addition to some basic privileges tests, this PR refactors the `server_query` function apparatus. A single `server_query` function replaces all helper functions including `multi_query`, `insert_query`, `server_query_with_user`, etc. This single interface removes a bunch of duplicate code in `query-server-common.bash` making everything easier to maintain. 

You now need to pass user and password to `server_query` which required changing everywhere in bats where `server_query` and all of its duplicates were called.

### CODE REVIEW

**Comment by timsehn** [comment]
I think I'm going remove the copy/pasta in `query-server-common.bash` because it's been bothering me. I need to make one function with the `server_query_with_user_password` interface and change all the call sites to that. There are like 100. It'll be fine.

**Comment by timsehn** [comment]
@Hydrocharged can review the privileges stuff. Not sure who should review `server_query`.

File: integration-tests/bats/remotes-sql-server.bats [hunk 1]
```diff
@@ -245,10 +239,9 @@ teardown() {
     dolt config --local --add sqlserver.global.dolt_replicate_heads unknown
     start_sql_server repo2
 
-    run server_query repo2 1 "show tables" "Table\n"
-    [ "$status" -eq 1 ]
-    [[ ! "$output" =~ "panic" ]]
+    server_query repo2 1 dolt "" "show tables" "" 1
     [[ "$output" =~ "remote not found: 'unknown'" ]] || false
+
```
Comment (max-hoffman) [inline integration-tests/bats/remotes-sql-server.bats:244]: nit: stray line

File: integration-tests/bats/sql-server-schema-fragments.bats [hunk 2]
```diff
@@ -1,86 +0,0 @@
-#!/usr/bin/env bats
-load $BATS_TEST_DIRNAME/helper/common.bash
-load $BATS_TEST_DIRNAME/helper/query-server-common.bash
```
Comment (max-hoffman) [inline integration-tests/bats/sql-server-schema-fragments.bats:3]: set head is deprecated now?

File: integration-tests/bats/helper/query-server-common.bash [hunk 3]
```diff
@@ -256,86 +200,33 @@ stop_sql_server() {
...
 
-# server_query connects to a running mysql server, executes a query and compares the results against what is expected.
-# In the event that the results do not match expectations, the python process will exit with an exit code of 1
+# server_query connects to a running mysql server,
+# executes a query (or list of queries separated by a `;`),
+# and compares the results against what is expected.
+#
+# If executing multiple queries,  separate the expected CSV values with a `;`.
+#
+# If you expect an exception, leave query results blank and add an additional
```
Comment (max-hoffman) [inline integration-tests/bats/helper/query-server-common.bash:209]: i like the expanded docstring, only thing i might add is example commands corresponding to the features you point out

**Review by max-hoffman — APPROVED** [review]
LGTM, didn't find many issues

File: integration-tests/bats/sql-server-schema-fragments.bats (hunk 2 above)
Comment (timsehn) [reply integration-tests/bats/sql-server-schema-fragments.bats:3]: Yeah. Tests didn't work so Aaron said delete em.

**Pull request merged**